### PR TITLE
add client ids to system state

### LINKED ISSUE #1562: vis: add client ids to system state
Currently, we just get the number of clients in each room. We want to be able to get the actual client ids for visualizing activity.

### PR DESCRIPTION
**Pull request by Victor-M-Giraldo**
- closes #1562

### CODE REVIEW

**Comment by Victor-M-Giraldo** [comment]
Still need to actually fix the tests.

File: packages/ott-vis-panel/src/components/TreeDisplay.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ function buildMonolithTrees(monoliths: Monolith[]): TreeNode[] {
 				id: room.name,
 				region: monolith.region,
 				group: "room",
-				children: Array.from({ length: room.clients }, (_, index) => {
+				children: Array.from({ length: room.clients.length }, (_, index) => {
 					return {
 						id: `${room.name}-${index}`,
```
Comment (dyc3) [inline packages/ott-vis-panel/src/components/TreeDisplay.tsx:70]: This needs to become the client id

File: packages/ott-vis-panel/src/components/TreeDisplay.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ function buildMonolithTrees(monoliths: Monolith[]): TreeNode[] {
 				id: room.name,
 				region: monolith.region,
 				group: "room",
-				children: Array.from({ length: room.clients }, (_, index) => {
+				children: Array.from({ length: room.clients.length }, (_, index) => {
```
Comment (dyc3) [inline packages/ott-vis-panel/src/components/TreeDisplay.tsx:68]: this `Array.from` hack can be removed in favor of a `.map` or something more readable.

**Review by dyc3 — COMMENTED** [review]
Looking good so far.

**Comment by Victor-M-Giraldo** [comment]
Don't know if the unit test is really necessary, but I think this ones good.

**Review by dyc3 — CHANGES_REQUESTED** [review]
This is good for `ott-vis-panel`, but `ott-collector` and `ott-balancer` still need to be updated accordingly.

Currently, it seems like the interop between the collector and the panel isn't being enforced in CI.

**Comment by dyc3** [comment]
You should rebase this PR for better type checking

**Comment by dyc3** [comment]
Whoops, I made some changes earlier today that are conflicting. Sorry about that, I can rebase this for you if you want

**Comment by Victor-M-Giraldo** [comment]
> Whoops, I made some changes earlier today that are conflicting. Sorry about that, I can rebase this for you if you want

Yeah, sure. I tried to do it myself. There is only one conflict but the incoming change is an older version of the `buildMonolithTrees` which I don't get why.

**Comment by Victor-M-Giraldo** [comment]
Since `countRoomClients` is returning a map of names to number of clients, I don't think I can access the id for the RegionView and GlobalView.

**Review by dyc3 — APPROVED** [review]
Yeah don't worry about global/region view. We're probably going to trash them.

**Comment by Victor-M-Giraldo** [comment]
Okay, but say if you did want to do that would you change `countRoomClients` to be like `Record<string, ClientState[]` or something.

**Comment by dyc3** [comment]
I would probably leave that count function as is and create a new one.

**Pull request merged**